rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -48,6 +49,15 @@ end
 local function balance()
   local backend = get_current_backend()
   local lb_alg = get_current_lb_alg()
+  local is_sticky = sticky.is_sticky(backend)
+
+  if is_sticky then
+    local upstream = sticky.get_upstream(backend)
+    if upstream ~= nil then
+      return upstream["host"], upstream["port"]
```

Comment:
Can you make this `get_endpoint` and actually return an endpoint object? This way we can unify balancer methods - i.e they will have consistent interface.

Reply:
So the shared dictionary stored a string `<ip>:<port>` which is split once retrieved. Creating an endpoint object seems unnecessary here.